### DISCUSSION TITLE
Blank page in production

### DISCUSSION THREAD

**Discussion by usmansagri** [remix-run/react-router#8879] [Q&A]
I have built a simple react app using `create-react-app` and `react-router-dom@6`  and when i build production it works fine when i run it using `serve -s build` and when i upload it on to my server i see only blank page.

And this happens only when i use `BrowserRouter`, `HashRouter` and `MemoryRouter` work just fine.

Please guide me how to build production app using `BrowserRouter` keeping in mind i will deploying it on linux server(hostinger.in) and in a subdirectory `domain.com/react-app`(!if .htaccess is required)

**Few things i tried**
1. Adding "homepage" : "./" or "homepage" : "." in `package.json`
2. Many .htaccess codes :')

You can see the hosted app [Here](https://sagri.co.in/captcha-generator/)

**Comment by mohameddjebloun** [comment]
Can you tell me if you fixed this bug ? And if you did can you please tell me how ?

**Comment by cuongnm2301** [comment]
Got same issue, any ideal to fix this

**Comment by Quanghihicoder** [comment]
Same
